cli/utils.go

```diff
@@ -251,5 +257,38 @@ func registerCLIExample(ctx context.Context, name, usage string) {
 	if exists {
 		panic("CLI example with the same name already exists: " + name)
 	}
-	registry.examples[name] = strings.ReplaceAll(usage, "\n", "")
+
+	if strings.Contains(usage, " | ") {
+		usageParts := strings.Split(usage, "|")
+		usage = usageParts[1]
+	}
+	registry.examples[name] = strings.ReplaceAll(usage, "\\\n", "")
+}
+
+func validateCLIArgs(cmd *cobra.Command, args []string) error {
+	cmd, args, err := cmd.Find(args)
+	if err != nil {
+		return err
+	}
+
+	if !cmd.Runnable() {
+		return fmt.Errorf("command isn't runnable: %s", cmd.Name())
+	}
+
+	flags := cmd.Flags()
+	flags.Parse(args)
+
+	remainingArgs := flags.Args()
+
+	if cmd.Args != nil {
+		if err := cmd.Args(cmd, remainingArgs); err != nil {
+			return err
+		}
+	}
+
+	if err := cmd.ValidateRequiredFlags(); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
question: How long does this validation take in your testing?

Reply:
I should add, this is only doing syntactic validation relative to the (sub) commands, defined/required flags, and expected number of arguments.

Its not actually *running* the command execution